web/ttyd-osc52/generate.py

```
@@ -0,0 +1,135 @@
+def load_provenance() -> dict[str, object]:
+    data = json.loads(PROVENANCE_PATH.read_text())
+    require_equal(data.get("ttyd_version"), SUPPORTED_TTYD_VERSION, "provenance ttyd version")
+    require_equal(configured_ttyd_version(), SUPPORTED_TTYD_VERSION, "repository ttyd version")
+    require_equal(data.get("compatibility_issue"), 174, "compatibility review issue")
+    return data
+
+
+def read_archive(path: Path | None, provenance: dict[str, object]) -> bytes:
+    if path is not None:
+        return path.read_bytes()
+    request = urllib.request.Request(str(provenance["archive_url"]), headers={"User-Agent": "remote-dev-ttyd-osc52-generator"})
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)